support-frontend/assets/pages/contributions-landing/components/StripeCardForm/StripeCardForm.jsx

```diff
@@ -0,0 +1,229 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React, { Component } from 'react';
+import { injectStripe, CardNumberElement, CardExpiryElement, CardCVCElement } from 'react-stripe-elements';
+import { connect } from 'react-redux';
+import type { State, Stripe3DSResult } from 'pages/contributions-landing/contributionsLandingReducer';
+import type { PaymentAuthorisation } from 'helpers/paymentIntegrations/readerRevenueApis';
+import { Stripe } from 'helpers/paymentMethods';
+import { type PaymentResult } from 'helpers/paymentIntegrations/readerRevenueApis';
+import { setCreateStripePaymentMethod,
+  setHandleStripe3DS,
+  setStripeCardFormComplete,
+  onThirdPartyPaymentAuthorised,
+  paymentFailure,
+  paymentWaiting,
+  type Action } from 'pages/contributions-landing/contributionsLandingActions';
+import { type ContributionType } from 'helpers/contributions';
+import type { ErrorReason } from 'helpers/errorReasons';
+import { logException } from 'helpers/logger';
+import { trackComponentLoad } from 'helpers/tracking/behaviour';
+
+// ----- Types -----//
+
+/* eslint-disable react/no-unused-prop-types */
+type PropTypes = {|
+  stripe: Object, // Available through the injectStripe 'Higher-Order Component'
+  onPaymentAuthorised: (PaymentAuthorisation) => Promise<PaymentResult>,
+  paymentFailure: (paymentError: ErrorReason) => Action,
+  contributionType: ContributionType,
+  setCreateStripePaymentMethod: ((email: string) => void) => Action,
+  setHandleStripe3DS: ((clientSecret: string) => Promise<Stripe3DSResult>) => Action,
+  paymentWaiting: (isWaiting: boolean) => Action,
+  setStripeCardFormComplete: (isComplete: boolean) => Action,
+|};
+
+const mapStateToProps = (state: State) => ({
+  contributionType: state.page.form.contributionType,
+});
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  onPaymentAuthorised:
+    (paymentAuthorisation: PaymentAuthorisation) =>
+      dispatch(onThirdPartyPaymentAuthorised(paymentAuthorisation)),
+  paymentFailure: (paymentError: ErrorReason) => dispatch(paymentFailure(paymentError)),
+  setCreateStripePaymentMethod: (createStripePaymentMethod: (email: string) => void) =>
+    dispatch(setCreateStripePaymentMethod(createStripePaymentMethod)),
+  setHandleStripe3DS: (handleStripe3DS: (clientSecret: string) => Promise<Stripe3DSResult>) =>
+    dispatch(setHandleStripe3DS(handleStripe3DS)),
+  setStripeCardFormComplete: (isComplete: boolean) =>
+    dispatch(setStripeCardFormComplete(isComplete)),
+  paymentWaiting: (isWaiting: boolean) =>
+    dispatch(paymentWaiting(isWaiting)),
+});
+
+type CardFieldState =
+  {| name: 'Error', errorMessage: string |} |
+  {| name: 'Incomplete' |} |
+  {| name: 'Complete' |};
+
+type CardFieldName = 'CardNumber' | 'Expiry' | 'CVC';
+
+type StateTypes = {
+  [CardFieldName]: CardFieldState,
+  currentlySelected: CardFieldName | null,
+};
+
+const fieldStyle = {
```

Comment:
I prefer styling react in this way but it breaks our convention - is it a requirement for stripe elements? do they not have constant ids or classes we can hook into?

Reply:
Indeed, the docs say:
> No CSS styles should be applied directly to anything Elements inserts inside your container DOM element. Style the container DOM element instead.